Bottom: 916a891d6ff56a53494db53da9b64a205620e2cd
Top:    629326d401d5bb02569d5c6ae014bb7116593a26
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-05-10 17:40:15 +0100

Improve dump_poly() output

Make it print each contour of a POLYAREA, and move the NEXT PLINE
and NEXT POLY messages inside the loops interating over these
structures.


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 2ffd7cb..d9f6ce4 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -142,20 +142,26 @@ pline_dump (VNODE * v)
 	       n->point[0], n->point[1], theState (v));
     }
   while ((v = v->next) != s);
-  fprintf (stderr, "NEXT PLINE\n");
 }
 
 static void
 poly_dump (POLYAREA * p)
 {
   POLYAREA *f = p;
+  PLINE *pl;
 
   do
     {
-      pline_dump (&p->contours->head);
+      pl = p->contours;
+      do
+        {
+          pline_dump (&pl->head);
+          fprintf (stderr, "NEXT PLINE\n");
+        }
+      while ((pl = pl->next) != NULL);
+      fprintf (stderr, "NEXT POLY\n");
     }
   while ((p = p->f) != f);
-  fprintf (stderr, "NEXT_POLY\n");
 }
 #endif
